core/src/main/java/org/apache/lucene/queries/BlendedTermQuery.java

```diff
@@ -363,15 +352,15 @@ public static BlendedTermQuery dismaxBlendedQuery(Term[] terms, final float[] bo
         return new BlendedTermQuery(terms, boosts) {
             @Override
             protected Query topLevelQuery(Term[] terms, TermContext[] ctx, int[] docFreqs, int maxDoc) {
-                DisjunctionMaxQuery query = new DisjunctionMaxQuery(tieBreakerMultiplier);
+                DisjunctionMaxQuery disMaxQuery = new DisjunctionMaxQuery(tieBreakerMultiplier);
                 for (int i = 0; i < terms.length; i++) {
-                    TermQuery termQuery = new TermQuery(terms[i], ctx[i]);
-                    if (boosts != null) {
-                        termQuery.setBoost(boosts[i]);
+                    Query query = new TermQuery(terms[i], ctx[i]);
```

Comment:
yes maybe, but it would look a bit messy cause the helper method would need too many arguments e.g. current_term, current_context, boosts. not sure it buys us much.